src/compiler/scala/tools/nsc/symtab/classfile/ClassfileParser.scala

```diff
@@ -648,13 +648,11 @@ abstract class ClassfileParser(reader: ReusableInstance[ReusableDataReader]) {
         case BOOL_TAG   => BooleanTpe
         case 'L' =>
           def processInner(tp: Type): Type = tp match {
-            case TypeRef(pre, sym, args) if (!sym.isStatic) =>
-              typeRef(processInner(pre.widen), sym, args)
-            case _ =>
-              tp
+            case TypeRef(_pre, _sym, _args) if !_sym.isStatic => typeRef(processInner(_pre.widen), _sym, _args)
+            case _ => tp
           }
           def processClassType(tp: Type): Type = tp match {
-            case TypeRef(pre, classSym, args) =>
+            case TypeRef(pre, typerefSym, args) =>
```

Comment:
this is where the usage is far away. So it's the combination of shadowing and distance. There ought to be a complexity measure that accounts for it.